ticketwriter/s3.go

```diff
@@ -0,0 +1,39 @@
+package ticketwriter
+
+import (
+	"bufio"
+	"github.com/goamz/goamz/aws"
+	"github.com/goamz/goamz/s3"
+	"net/http"
+
+	"os"
+)
+
+// uploadToS3 uploads the contents of tickWrt.currenFileName to S3
+func (tickWrt *TicketWriter) uploadToS3() {
```

Comment:
I have an idea for improving this function a bit. You could use the HEAD method of the key after it was successfully uploaded and verify that the ETag matches the MD5 hash of the file locally. It would give you some reassurance of the data actually making it to S3 intact.